x-pack/plugins/osquery/public/shared_components/osquery_action/index.tsx

```diff
@@ -61,7 +57,10 @@ const OsqueryActionComponent: React.FC<OsqueryActionProps> = ({
     return emptyPrompt;
   }
 
-  if (!(permissions.runSavedQueries || permissions.writeLiveQueries)) {
+  if (
+    (!permissions.runSavedQueries || !permissions.readSavedQueries) &&
+    !permissions.writeLiveQueries
+  ) {
```

Comment:
Would be good to add a unit test for this so it does not break again in the future.

Reply:
I am working on it, however if you don't mind it's gonna be an e2e test. Thanks!

Reply:
Even better :)